### PR TITLE
NS-1025 Transactional Producer Support

### PR DESCRIPTION
**Pull request by haroldb1**
Add support for transactional producer - either all messages are published, or none.

### CODE REVIEW

File: akka/src/main/scala/cakesolutions/kafka/akka/KafkaProducerActor.scala [hunk 1]
```diff
@@ -142,11 +234,25 @@ private class KafkaProducerActor[K, V](
...
 
   private def sendMany(records: Records) = Future.sequence(records.map(send))
 
+  private def sendManyTransactional(records: Records): Future[Iterable[RecordMetadata]] =
+    Future
+      .fromTry(Try(producer.beginTransaction()))
+      .flatMap(_ =>
+        (for {
+          results <- sendMany(records)
+          _ = Future.fromTry(Try(producer.commitTransaction()))
```
Comment (regiskuckaertz) [inline akka/src/main/scala/cakesolutions/kafka/akka/KafkaProducerActor.scala]: This a bit confusing, so it is fine to commit the transaction but lose the error? If not I'd suggest
```
_ <- Future.fromTry(Try(producer.commitTransaction()))
```

File: akka/src/main/scala/cakesolutions/kafka/akka/KafkaProducerActor.scala [hunk 2]
```diff
@@ -142,11 +234,25 @@ private class KafkaProducerActor[K, V](
...
+        } yield results).recoverWith { case sendThrowable =>
```
Comment (regiskuckaertz) [inline akka/src/main/scala/cakesolutions/kafka/akka/KafkaProducerActor.scala:248]: Note here the transaction may not have started in the first place! i.e. if `beginTransaction` failed.

Comment (haroldb1) [reply akka/src/main/scala/cakesolutions/kafka/akka/KafkaProducerActor.scala:248]: it shouldn't get here is begin fails.  thats why the result of beginTransaction is flatMapped here.

File: akka/src/main/scala/cakesolutions/kafka/akka/KafkaProducerActor.scala (hunk 1 above)
Comment (haroldb1) [reply akka/src/main/scala/cakesolutions/kafka/akka/KafkaProducerActor.scala]: fixed

**Review by regiskuckaertz — APPROVED** [review]
Nice!

**Pull request merged**